canary/generator/download/download_split_by.py

```diff
@@ -0,0 +1,68 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from collections import defaultdict
+from multiprocessing import Pool
+from itertools import product
+import json
+from json import JSONDecodeError
+from ast import literal_eval
+import os
+import sys
+
+import requests
+import numpy as np
+
+from .query import build_query_string
+
+
+def _get_data(arglist):
+    dates_for_each, m, ch, v, os, app, ar = arglist
+    q = build_query_string(dates_for_each, m, ch, v, os, application=app, architecture=ar)
+    data = requests.get(q).text
+    try:
+        data = json.loads(data)
+    except JSONDecodeError:
+        print('bad metric: ', m)
+        return
+    data['channel'] = ch
+    data['metric'] = m
+    data['version'] = v
+    data['os'] = os
+    data['application'] = app
+    data['architecture'] = ar
+    return data
+
+
+if __name__ == '__main__':
+    pool = Pool(20)
+    channels = requests.get(
+        'https://aggregates.telemetry.mozilla.org/aggregates_by/build_id/channels/').text
+    channels = literal_eval(channels)
+    ch = 'nightly'
+    dates_versions = requests.get(
+        'https://aggregates.telemetry.mozilla.org/'
+        'aggregates_by/build_id/channels/' + ch + '/dates/').text
+    dates_versions = literal_eval(dates_versions)
+    dates_for_each = defaultdict(list)
+    for item in dates_versions:
+        dates_for_each[item['version']].append(item['date'])
+    last_versions = sorted(dates_for_each.keys())[-1:]
+    new_versions = {v: dates_for_each[v] for v in last_versions}
+    for v in new_versions.keys():
+        print(v)
+        options = requests.get(
+            'https://aggregates.telemetry.mozilla.org/'
+            'filters/?channel=' + ch + '&version=' + v).text
+        options = literal_eval(options)
+        options['os'] = np.unique([os.split(',')[0] for os in options['os']])
+        for m in options['metric']:
+            whole_data_for_metric = []
+            ops = list(product([dates_for_each[v]], [m], [ch], [v], options['os'],
+                               options['application'], options['architecture']))
+            for data in pool.imap_unordered(_get_data, ops):
```

Comment:
Can you please document how the different dimensions are saved? I mean, if I'm interested to a specific architecture, how can I look up that data from the dumped dataset?